### PR TITLE
fix: enhance label expression parser to support AND, OR, and NOT keywords

### LINKED ISSUE #1508: Compound label expressions (AND operator) rejected as invalid
## Description

The CLI and server reject compound label expressions using the `AND` operator, despite the syntax being documented in `wanaku man label-expression`.

## Steps to Reproduce

1. Start the local Wanaku stack
2. Register tools with labels:
   ```
   wanaku tools add --name tool-a --uri "http://a" --type http --namespace public --description "A"
   wanaku tools label add --name tool-a --label env=prod --label tier=frontend
   ```
3. Filter with a compound expression:
   ```
   wanaku tools list -e "env=prod AND tier=frontend" --plain
   ```

## Expected Behavior

Returns only tools matching both labels (tool-a).

## Actual Behavior

Server returns: `Invalid label expression: env=prod AND tier=frontend`

## Additional Context

Single-label expressions (e.g., `-e "env=prod"`) work correctly. The `AND` operator is documented in the label expression help page (`wanaku man label-expression`).

### PR DESCRIPTION
**Pull request by matheusandre1**
Closes: #1508

## Summary by Sourcery

Extend label expression parsing to recognize AND/OR/NOT as logical operators alongside existing symbols, and update docs and tests accordingly.

New Features:
- Support the AND keyword as a logical conjunction in label expressions.
- Support the OR keyword as a logical disjunction in label expressions.
- Support the NOT keyword as a logical negation in label expressions.

Enhancements:
- Improve tokenizer to distinguish logical operator keywords from identifier values based on surrounding whitespace.

Documentation:
- Document support for the AND keyword as an alternative to the & operator in label expressions.

Tests:
- Add unit tests covering AND, OR, and NOT keyword usage, mixed keyword/symbol expressions, parenthesized keyword expressions, and keywords used as label values.

### CODE REVIEW

File: apps/wanaku-router-backend/src/main/java/ai/wanaku/backend/core/mcp/util/LabelExpressionParser.java [hunk 1]
```diff
@@ -16,13 +16,14 @@
...
+ *   <li>"(category=weather OR category=news) AND !action=forecast"</li>
  * </ul>
  * <p>
  * Grammar:
  * <pre>
  * expression  ::= orExpr
  * orExpr      ::= andExpr ('|' andExpr)*
- * andExpr     ::= notExpr ('&amp;' notExpr)*
+ * andExpr     ::= notExpr ('&amp;' notExpr | 'AND' notExpr)*
  * notExpr     ::= '!' notExpr | primary
```
Comment (orpiske) [inline apps/wanaku-router-backend/src/main/java/ai/wanaku/backend/core/mcp/util/LabelExpressionParser.java]: The grammar comment now reflects the `AND` keyword, but the `orExpr` and `notExpr` rules still show only the symbol forms. Consider updating all three for consistency:

```suggestion
 * orExpr      ::= andExpr ('|' andExpr | 'OR' andExpr)*
 * andExpr     ::= notExpr ('&amp;' notExpr | 'AND' notExpr)*
 * notExpr     ::= ('!' | 'NOT') notExpr | primary
```

File: apps/wanaku-router-backend/src/main/java/ai/wanaku/backend/core/mcp/util/LabelExpressionParser.java [hunk 2]
```diff
@@ -256,6 +262,32 @@ private Predicate<LabelsAwareEntity<?>> parseComparison() throws LabelExpression
 
     // Helper methods
 
+    private static TokenType keywordTokenType(String text, int nextPosition, String expression) {
```
Comment (orpiske) [inline apps/wanaku-router-backend/src/main/java/ai/wanaku/backend/core/mcp/util/LabelExpressionParser.java:272]: Nice keyword boundary detection — this correctly prevents `label=AND` from being treated as an operator.

One edge case to consider: `NOT(deprecated=true)` and `env=prod AND(tier=frontend)` won't recognize the keyword because `(` isn't treated as a valid boundary. Since `!(...)` and `&(...)` work fine with the symbol operators, this isn't blocking, but consider also accepting `(` and `)` as keyword boundaries for a more intuitive experience. A small helper like `isKeywordBoundary(idx, expression)` that checks whitespace, parens, start-of-string, and end-of-string would cover it.

File: apps/wanaku-cli/src/main/resources/docs/LABEL_EXPRESSIONS.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ wanaku tools list -l "environment!=production"
 
 | Operator | Description | Example                             |
 |----------|-------------|-------------------------------------|
-| `&`      | Logical AND | `category=weather & version=2.0`    |
+| `&` or `AND` | Logical AND | `category=weather AND version=2.0` |
```
Comment (orpiske) [inline apps/wanaku-cli/src/main/resources/docs/LABEL_EXPRESSIONS.md]: The AND row now shows both symbol and keyword forms — nice! For consistency, consider also updating the OR and NOT rows below to show both the symbol and keyword alternatives, following the same pattern.

Also consider adding a note that keywords are case-sensitive (uppercase only).

**Review by orpiske — COMMENTED** [review]
Thanks for the contribution, @matheusandre1! The tokenizer-level approach is well-designed — resolving keywords to the same `TokenType` as symbol operators keeps the recursive descent parser clean and avoids code duplication.

A few suggestions below, mostly around documentation consistency. The core fix for #1508 is solid.

**Note on case sensitivity:** The `keywordTokenType` method uses `equals()` (exact match), so only uppercase `AND`, `OR`, `NOT` are recognized — `and`, `Or`, etc. are silently treated as identifiers. This is a reasonable design choice, but worth documenting in `LABEL_EXPRESSIONS.md` so users know upfront.

This review does not replace specialized tools like CodeRabbit, Sourcery, or SonarCloud.

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._

**Review by orpiske — APPROVED** [review]
Clean implementation — the keyword boundary detection logic is well thought out and correctly handles all edge cases I checked (keywords as label values, keywords adjacent to parentheses, keywords embedded in longer identifiers). Test coverage is thorough with 10 new test methods.

One minor note: the branch is named `issue1508` — the project convention is `ci-fix-<ISSUE_NUMBER>`. Not blocking.

**Note:** This is a conventions-and-correctness review. It does not replace specialized review tools (CodeRabbit, Sourcery) or static analyzers.

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._
